src/main/java/io/confluent/connect/jdbc/dialect/SybaseDatabaseDialect.java

```diff
@@ -0,0 +1,290 @@
+    // and JDBC's schema to represent the owner (e.g., "dbo")
+    return true;
+  }
+
+  @Override
+  protected String currentTimestampDatabaseQuery() {
+    return "select getdate()";
+  }
+
+  @Override
+  protected String checkConnectionQuery() {
+    return "SELECT 1";
+  }
+
+  @Override
+  protected String getSqlType(SinkRecordField field) {
+    if (field.schemaName() != null) {
+      switch (field.schemaName()) {
+        case Decimal.LOGICAL_NAME:
+          return "decimal(38," + field.schemaParameters().get(Decimal.SCALE_FIELD) + ")";
+        case Date.LOGICAL_NAME:
+          return "date";
+        case Time.LOGICAL_NAME:
+          return "time";
+        case Timestamp.LOGICAL_NAME:
+          return "datetime";
+        default:
+          // pass through to normal types
+      }
+    }
+    switch (field.schemaType()) {
+      case INT8:
+        // 'tinyint' has a range of 0-255 and cannot handle negative numbers
+        return "smallint";
+      case INT16:
+        return "smallint";
+      case INT32:
+        return "int";
+      case INT64:
+        return "bigint";
+      case FLOAT32:
+        return "real";
+      case FLOAT64:
+        return "float";
+      case BOOLEAN:
+        if (field.isOptional()) {
+          return "tinyint"; // Sybase does not allow 'bit' to be nullable
+        } else {
+          return "bit";
+        }
+      case STRING:
+        if (field.isPrimaryKey()) {
+          // Could always use 'text', except columns of type 'text', 'image' and 'unitext'
+          // cannot be used in indexes. Also, 2600 is the max allowable size of an index,
+          // so use something smaller if multiple columns are to be used in the index.
+          return "varchar(512)";
+        } else {
+          return "text";
+        }
+      case BYTES:
+        return "image";
+      default:
+        return super.getSqlType(field);
+    }
+  }
+
+  protected boolean maybeBindPrimitive(
+      PreparedStatement statement,
+      int index,
+      Schema schema,
+      Object value
+  ) throws SQLException {
+    // First handle non-standard bindings ...
+    switch (schema.type()) {
+      case INT8:
+        if (value instanceof Number) {
+          statement.setShort(index, ((Number) value).shortValue());
+          return true;
+        }
+        break;
+      default:
+        break;
+    }
+    return super.maybeBindPrimitive(statement, index, schema, value);
+  }
+
+  @Override
+  public void applyDdlStatements(
+      Connection connection,
+      List<String> statements
+  ) throws SQLException {
+    boolean autoCommit = connection.getAutoCommit();
+    if (!autoCommit) {
+      connection.setAutoCommit(true);
+    }
+    try (Statement statement = connection.createStatement()) {
+      for (String ddlStatement : statements) {
+        statement.executeUpdate(ddlStatement);
+      }
+    } finally {
+      connection.setAutoCommit(autoCommit);
+    }
```

Comment:
Sybase is one of those DBMSes that does everything a bit differently. The JDBC sink connector turns off autocommit, so any DDL statements it executes is followed by an explicit commit. Of course Sybase doesn't like this, so here we have to temporarily turn on autocommit, apply the DDL statements, and then turn back off autocommit.